### PR TITLE
riscv: add vfork and ptregs_jmp

### PR DESCRIPTION
**Pull request by nastya-nizharadze**
Add vfork and ptregs_jmp. Work on #1047

### CODE REVIEW

File: templates/riscv/qemu-sifive-u/mods.config [hunk 1]
```diff
@@ -17,16 +18,40 @@ configuration conf {
...
+	include embox.compat.posix.fs.close
+
 	include embox.kernel.task.resource.errno
 
 	include embox.compat.libc.str
+	include embox.framework.LibFramework
 
 	include embox.test.stdlib.setjmp_test
 	include embox.test.kernel.timer_test
-	include embox.test.kernel.thread.thread_test
```
Comment (alexkalmuk) [inline templates/riscv/qemu-sifive-u/mods.config]: Why you deleted thread_test from config? Does it work normally?

File: templates/riscv/qemu-sifive-u/mods.config [hunk 2]
```diff
@@ -17,16 +18,40 @@ configuration conf {
...
-	include embox.test.hal.context_switch_test(stack_sz=0x40)
+	include embox.test.posix.vfork_test
+	include embox.test.hal.context_switch_test(stack_sz=0x3c)
```
[the remark this answers is not in the document]
Comment (nastya-nizharadze) [reply templates/riscv/qemu-sifive-u/mods.config]: Indeed. Done.

File: templates/riscv/qemu-sifive-u/mods.config (hunk 1 above)
Comment (nastya-nizharadze) [reply templates/riscv/qemu-sifive-u/mods.config]: Yes, it does. Brought it back.

Comment (alexkalmuk) [reply templates/riscv/qemu-sifive-u/mods.config]: I got the following:

```
Embox kernel start
	unit: initializing embox.mem.phymem: 
			start=0x80069680 end=0xa0000000 size=536439168
		done
runlevel: init level is 0
	unit: initializing embox.kernel.task.task_resource: done
	unit: initializing embox.kernel.task.task_table: done
	unit: initializing embox.kernel.task.kernel_task: done
	unit: initializing embox.driver.interrupt.riscv_intc: done
	unit: initializing embox.kernel.sched.sched: done
	unit: initializing embox.arch.riscv.kernel.interrupt: done
	unit: initializing embox.driver.clock.riscv_clk: done
	unit: initializing embox.kernel.time.timer_handler: done
	unit: initializing embox.kernel.time.kernel_time: done
runlevel: init level is 1
	test: running embox.test.hal.context_switch_test . done
	test: running embox.test.posix.vfork_test .... done
	test: running embox.test.kernel.thread.thread_test .

  ______
 |  ____|                                            __          __
 | |___  _ __ ___            ____  ____  ____  _____/ /   _____ / /
 |  ___|| '_ ` _ \          / __ \/ __ \/ __ \/ ___/ /   |_____| |
 | |____| | | | | |_ _ _   / /_/ / /_/ / /_/ (__  )_/    |_____| |
 |______|_| |_| |_(_|_|_)  \____/\____/ .___/____(_)           | |
                                     /_/                        \_\
 ASSERTION FAILED on CPU 0
	at src/kernel/thread/core.c:272
	in function thread_delete

!t->schedee.ready
```

Comment (alexkalmuk) [reply templates/riscv/qemu-sifive-u/mods.config]: Launched like `qemu-system-riscv32 -M sifive_u -kernel ./build/base/bin/embox -m 512 -nographic`.

Comment (alexkalmuk) [reply templates/riscv/qemu-sifive-u/mods.config]: Can you please check that?

Comment (nastya-nizharadze) [reply templates/riscv/qemu-sifive-u/mods.config]: I launch qemu in thus way: `qemu-system-riscv32 -nographic -machine sifive_u -kernel build/base/bin/embox`
qemu 4.1.0

Comment (alexkalmuk) [reply templates/riscv/qemu-sifive-u/mods.config]: Ok, thank you, I have the same qemu 4.1.0.

**Comment by Deryugin** [comment]
A have no output at all on QEMU 4.2.0

**Comment by alexkalmuk** [comment]
> A have no output at all on QEMU 4.2.0

Yes, we already know :)

**Comment by Deryugin** [comment]
Please, don't forget to put "in progress" label to avoid excessive checks if it's actually in progress :)

**Comment by alexkalmuk** [comment]
> Please, don't forget to put "in progress" label to avoid excessive checks if it's actually in progress :)

Ok, we were discussing issue with thread_test failure with Nastya (should be fixed soon) and I didn't mention about 4.2.0 bug, sorry.

Anyway, I think bug related to 4.2.0 should be another issue. What do you think?

**Comment by Deryugin** [comment]
Yes, I think that it's a good idea to make issue for this

**Comment by alexkalmuk** [comment]
@Deryugin  I've created the issue about 4.2.0. Thank you for verifying.

File: templates/riscv/qemu-sifive-u/mods.config (hunk 1 above)
Comment (alexkalmuk) [reply templates/riscv/qemu-sifive-u/mods.config]: BTW we investigated that the problem is in runlevels..

Comment (nastya-nizharadze) [reply templates/riscv/qemu-sifive-u/mods.config]: The problem was in runlevels. I fixed it.

Comment (alexkalmuk) [reply templates/riscv/qemu-sifive-u/mods.config]: Nice! Finally, it works for me. @Deryugin If you want you can try too on qemu 4.1.0 (without auto_qemu)

**Pull request merged**